### PR TITLE
adding test for Issue 1497

### LINKED ISSUE #1497: In case of a 400 response due to a validation error, include the validation error in error.message
This is a follow up for https://github.com/probot/probot/issues/1007#issuecomment-538772566

### PR DESCRIPTION
**Pull request by ram-nad**
fixes #1497

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -11,3 +11,5 @@ dist/
 
 # cypress
 cypress/
+
+/.vscode/
```
Comment (gr2m) [inline .gitignore]: Please remove that, it's unrelated to the project. You should have a global gitignore on your system that ignores the `.vscode` folder everywhere.

File: test/issues/1497-include-error-message-on-validation-error.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        enforce_admins: true,
+        required_pull_request_reviews: {
+          required_approving_review_count: 1,
+          dismiss_stale_reviews: true,
+          require_code_owner_reviews: true,
+          dismissal_restrictions: { users: [], teams: [] }
+        },
+        restrictions: { users: [], teams: [] },
+      })
+      .reply(200, {
```
Comment (gr2m) [inline test/issues/1497-include-error-message-on-validation-error.js]: The response code is `403`, not `200`.

File: test/issues/1497-include-error-message-on-validation-error.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+      required_status_checks: { strict: true, contexts: ["wip"] },
+      enforce_admins: true,
+      required_pull_request_reviews: {
+         required_approving_review_count: 1,
+         dismiss_stale_reviews: true,
+         require_code_owner_reviews: true,
+         dismissal_restrictions: { users: [], teams: [] }
+      },
+      restrictions: { users: [], teams: [] }
+    });
```
Comment (gr2m) [inline test/issues/1497-include-error-message-on-validation-error.js]: You still need to check that `octokit.repos.updateBranchProtection()` rejects and that the error includes the error message. You can check that `error.message` is `Validation failed: Only organization repositories can have users and team restrictions`

**Review by gr2m — CHANGES_REQUESTED** [review]

**Comment by ram-nad** [comment]
sure sir. Thanks.

Sir does this project use prettier?

**Comment by ram-nad** [comment]
Sir sorry for messing things up. I had to reset my branch. And commit changes again.

**Review by gr2m — COMMENTED** [review]
can you rename the faile to `test/issues/1497-include-error-message-on-validation-error-test.js`? Without the `-test.js` ending it will not be run by Mocha, which is why the test is currently not shown as failing

**Comment by ram-nad** [comment]
sir, is my code correct or the error is expected?

**Comment by gr2m** [comment]
Yes, an error is expected, you are creating a test case to reproduce the problem reporting in #1479.

But the error is not what we are looking for: https://github.com/octokit/rest.js/runs/252822246#step:7:256

It looks like the mock you defined does not match the request that is being sent. Can you look into that?

**Comment by ram-nad** [comment]
yes sir i will check again

**Comment by ram-nad** [comment]
Sir, I fixed the mock server. I had a question, why is accept header getting `machine-man` preview by default. Also sir i think that `application/vnd.github.v3+json` should also be added by default. Thank You.

**Comment by gr2m** [comment]
> I had a question, why is accept header getting machine-man preview by default. 

We can remove that. The only relevant preview is `luke-cage`, the others can be removed

> Also sir i think that application/vnd.github.v3+json should also be added by default.

If a preview header is set, then `application/vnd.github.v3+json` doesn't need to be set, too

**Comment by gr2m** [comment]
> I had a question, why is accept header getting machine-man preview by default

Oh I see, you don't set the `machine-man` preview yourself. That is very odd, it should not be set automatically, looks like a separate bug. Feel free to create a follow up issue / pull request, once we are don with this :)

The test error now says what we wanted:

```
      + expected - actual

      -Validation Failed
      +Validation failed: Only organization repositories can have users and team restrictions
```

Next up is changing the code to make the test pass. Would you like to give that a try? It's possible that it will need to be fixed in https://github.com/octokit/request.js/, I'd have to investigate myself

**Comment by ram-nad** [comment]
Sure sir i would like to try. Do we need to we need to change 3 files like `@octokit/request/dist-src`, `@octokit/request/dist-node`, `@octokit/request/dist-web` ?

**Comment by gr2m** [comment]
I would leave this pull request as is for now. 

Start a new pull request for https://github.com/octokit/request.js/, add a test similar to this one https://github.com/octokit/request.js/blob/0fc38c0fc0da22556fb348500b5c5844650e52e5/test/request.test.ts#L445-L484 but with the request you did in here.

`octokit.repos.updateBranchProtection(parameters)` is just an alias for `request("PUT /repos/:owner/:repo/branches/:branch/protection", parameters)`

Once we create the test and fix it in `@octokit/request`, we can update the `@octokit/request` version as part of this pull request and it the test should pass

**Comment by ram-nad** [comment]
Okk sir.

**Comment by ram-nad** [comment]
Sir would i use nock for mocking or `fetchMock`

**Comment by gr2m** [comment]
fetchMock in https://github.com/octokit/request.js, try to follow the way the other tests are done

**Comment by ram-nad** [comment]
sure sir. Sir there are certain type matching errors in the request.test.ts file. What should I do? Most of them are related to Headers type being incompatible.

**Comment by gr2m** [comment]
start a PR with what you have, I'll have a look tomorrow

**Comment by ram-nad** [comment]
okk thanks sir.

**Comment by ram-nad** [comment]
Sir, this `error-test` is failing. Should i  update that test also?

**Review by gr2m — APPROVED** [review]
Perfect, thanks @ram-nad!

**Comment by gr2m** [comment]
Hmm the tests are still failing, can you have a look what might be going on?

**Comment by ram-nad** [comment]
Yes sir. Actually a previously written test depends upon `error.message` being `Validation failed` instead of the complete message with reasons that it is now having.

**Comment by gr2m** [comment]
Can you adapt the other test?

**Review by gr2m — APPROVED** [review]
Thanks 👍

**Pull request merged**